client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -192,8 +192,8 @@ function mediaButton( editor ) {
 
 				// When merging, allow any updated field to be used if it doesn't
 				// already exist in the current markup, but otherwise only force
-				// update ID and URL attributes to their new values
-				const merged = assign( {}, media, current.media, pick( media, 'ID', 'URL' ), {
+				// update ID, URL, width and height attributes to their new values
+				const merged = assign( {}, media, current.media, pick( media, 'ID', 'URL', 'width', 'height' ), {
```

Comment:
Rebased with master since I've deployed `restore image` feature a few minutes ago. The restored image isn't updated into the post content.